v1/js/c.js

```diff
@@ -81,9 +81,18 @@ var c = {
         b:[[7,3]],
       },
       canMove:function(x1,y1,x2,y2){
-        if(chess.getPieceAt(x2,y2)){
-          if(chess.getPieceAt(x2,y2).team == this.team){
-            return false;
+        var destPiece = chess.getPieceAt(x2,y2);
+        if(destPiece){
+          if(destPiece.team == this.team){
+            console.log(destPiece.id);
+            if (destPiece.id == 'rook' &&   !(chess.getPieceAt(x1,y1).hasMoved() || destPiece.hasMoved())){
+              if ((y1< y2 && chess.isClearPath(x1,y1,x2,y2-1))||(y1> y2 && chess.isClearPath(x1,y1,x2,y2+1))) {
+              return true;
+              
+              }
+            }else{
+              return false;
+            }
           }
```

Comment:
I'm assuming this relates to castling?

Reply:
Castling is the act of moving the king and one of the rooks from the corner to the middle as a way of defending the King. So if they drag their king to one of the rooks which is still in the corner and the path is clear they will both move to be swapped on the tiles between them.